link.py

```diff
@@ -0,0 +1,115 @@
+from collections import deque
+'''
```

Comment:
Docstring for the class should be below "class Link(object)". Also docstring should be triple double quotes instead of single quotes